### PR TITLE
[Concept Entry] AI: Self-Supervised Learning

### LINKED ISSUE #5971: [Concept Entry] AI: Self-Supervised Learning
### Reason/inspiration (optional)

We would like a new entry on the `Self-Supervised Learning` concept under AI. The entry should go in a new file under `docs/content/ai/concepts/machine-learning/terms/self-supervised-learning/self-supervised-learning.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by danitellini**
### Description

Topic: AI: Self-Supervised Learning
File added: `docs/content/ai/machine-learning/terms/self-supervised-learning/self-supervised-learning.md`

### Issue Solved

Closes #5971 

### Type of Change

- Adding a new entry
- ~Editing an existing entry (fixing a typo, bug, issues, etc)~
- ~Updating the documentation~

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @danitellini  😄 
The entry looks good for a second round of review! 🚀

File: content/ai/concepts/machine-learning/terms/self-supervised-learning/self-supervised-learning.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/intermediate-machine-learning-skill-path'
+---
+
+**Self-Supervised Learning (SSL)** is a machine learning approach that bridges the gap between supervised and unsupervised learning. It enables models to learn meaningful representations from unlabeled data by deriving supervision signals through pretext tasks.
+
+SSL is widely applied in domains such as natural language processing (NLP), computer vision, and speech recognition.
+
+## Syntax
```
Comment (avdhoottt) [inline content/ai/concepts/machine-learning/terms/self-supervised-learning/self-supervised-learning.md:21]:
```suggestion
## Pipeline
```

File: content/ai/concepts/machine-learning/terms/self-supervised-learning/self-supervised-learning.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  loss.backward()
+  optimizer.step()
+
+print("Self-supervised model training complete!")
+```
+
+If the code runs successfully, the output will be:
+
+```shell
+Self-supervised model training complete!
```
Comment (avdhoottt) [inline content/ai/concepts/machine-learning/terms/self-supervised-learning/self-supervised-learning.md:97]:
```suggestion
Self-supervised model training complete!
```

```

Comment (avdhoottt) [inline content/ai/concepts/machine-learning/terms/self-supervised-learning/self-supervised-learning.md:97]:
```suggestion
```
